Zend/zend_compile.h

```diff
@@ -299,6 +299,9 @@ typedef struct _zend_property_info {
 	zend_string *name;
 	zend_string *doc_comment;
 	zend_class_entry *ce;
+	zend_uchar type;
+	zend_string *type_name;
```

Comment:
It might be possible to merge all 3 fields into single union using "bit stealling" technique, but this may be done separately and uniformly for all kind of type hints.